app_potato/src/telemetry.c

```diff
@@ -0,0 +1,93 @@
+// Self Include
+#include "potato.h"
+
+// Launch Includes
+#include <launch_core/dev/sensor.h>
+#include <launch_core/conversions.h>
+
+// Zephyr Includes
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+
+#define TELEMETRY_STACK_SIZE 512
+
+LOG_MODULE_REGISTER(telemetry);
+
+// Threads
+static void telemetry_read_task(void*);
+K_THREAD_DEFINE(telem_read_thread, TELEMETRY_STACK_SIZE, telemetry_read_task, NULL, NULL, NULL, K_PRIO_PREEMPT(20), 0,
+                1000);
+
+static void telemetry_processing_task(void*);
+K_THREAD_DEFINE(telem_process_thread, TELEMETRY_STACK_SIZE, telemetry_processing_task, NULL, NULL, NULL,
+                K_PRIO_PREEMPT(20), 0,
+                1000);
+
+// Timers
+K_TIMER_DEFINE(lps22_timer, NULL, NULL);
+
+// Queues
+K_MSGQ_DEFINE(raw_telem_processing_queue, sizeof(potato_raw_telemetry_t), 16, 1);
+
+// Global Variables
+float boost_detection_altitude = -0xFFFF;
+
+// External Variables
+extern struct k_msgq logging_queue;
+extern bool logging_enabled;
+
+static void convert_raw_telemetry(potato_raw_telemetry_t* raw_telem, potato_telemetry_t* telem) {
+    telem->timestamp = raw_telem->timestamp;
+    telem->altitude = l_altitude_conversion(raw_telem->lps22_data.pressure, raw_telem->lps22_data.temperature);
+
+    // TODO: Update
```

Comment:
Update what?

Reply:
Need to convert raw analog to a digital value